kubernetes/data_source_kubernetes_ingress.go

```diff
@@ -0,0 +1,27 @@
+package kubernetes
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func dataSourceKubernetesIngress() *schema.Resource {
+	s := resourceKubernetesIngress().Schema
+	s["metadata"] = namespacedMetadataSchema("ingress", false)
+	s["spec"].Computed = true
+	s["spec"].Required = false
+	return &schema.Resource{
+		Read:   dataSourceKubernetesIngressRead,
+		Schema: s,
+	}
+}
+
+func dataSourceKubernetesIngressRead(d *schema.ResourceData, meta interface{}) error {
+	om := meta_v1.ObjectMeta{
+		Namespace: d.Get("metadata.0.namespace").(string),
```

Comment:
For new code, we're trying to use this new syntax, since the old syntax is deprecated in the SDK:

```suggestion
		metadata := expandMetadata(d.Get("metadata").([]interface{}))
		Namespace: metadata.Namespace,
```